README.md

```diff
@@ -1809,6 +1811,18 @@ Replace 1000 by `exactCount` if configured differently.
 
 RMB is aware of the [metadata.schema](https://github.com/folio-org/raml/blob/raml1.0/schemas/metadata.schema). When a request (POST / PUT / PATCH) comes into an RMB module, RMB will check if the passed-in JSON's schema declares a reference to the metadata schema. If so, RMB will populate the JSON with a metadata section with the current user and the current time. RMB will set both update and create values to the same date/time and to the same user, as accepting this information from the request may be unreliable. The module should persist the creation date and the created by values after the initial POST. For an example of this using SQL triggers see [metadata.ftl](https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/main/resources/templates/db_scripts/metadata.ftl). Add [withMetadata to the schema.json](https://github.com/folio-org/raml-module-builder#the-post-tenant-api) to create that trigger.
 
+## Optimistic Locking
+
+RMB supports optimistic locking. By default it is disabled. Module developer can enable it by adding attribute `withOptimisticLocking` to the table definition in schema.json. The available options are listed below. When either `failOnConflict` or `logOnConflict` attribute are specified, a database trigger will be created to auto populate/update `_version` field in json on insert/update. Note, `_version` field has to be defined in json to make this work. The version conflict error will be reported as 419 HTTP response code if 419 is defined in RAML, otherwise, it will fall back to use 400 HTTP response code.
+
+    * `off` Optimistic Locking is disabled. The trigger is removed if it existed before.
+    * `failOnConflict` Optimistic Locking is enabled. Version conflict will fail the transaction
+    * `logOnConflict` Optimistic Locking is enabled. Version conflict info is logged as a warning message with a customized SQL error code 23F09. A sample log entry looks like below:
+
+```
+Backend notice: severity='WARNING', code='23F09', message='Cannot update record 57db089f-18e4-7815-55d5-4cc6607e9059 because it has been changed: Stored _version is 1, _version of request is "2"' ...
```

Comment:
Showing an example log entry is a good idea.
The version should show the common case where the request version is older than the stored version:
```suggestion
Backend notice: severity='WARNING', code='23F09', message='Cannot update record 57db089f-18e4-7815-55d5-4cc6607e9059 because it has been changed: Stored _version is 2, _version of request is "1"' ...
```

Reply:
> Showing an example log entry is a good idea.
> The version should show the common case where the request version is older than the stored version:

Good catch. Done.